src/uct/sm/mm/base/mm_iface.c

```diff
@@ -533,16 +575,29 @@ static UCS_CLASS_INIT_FUNC(uct_mm_iface_t, uct_md_h md, uct_worker_h worker,
     self->config.fifo_size         = mm_config->fifo_size;
     self->config.fifo_elem_size    = mm_config->fifo_elem_size;
     self->config.seg_size          = mm_config->seg_size;
+    self->config.fifo_max_poll     = ((mm_config->fifo_max_poll == UCS_ULUNITS_AUTO) ? 16 :
+                                      /* trim by the maximum value for unsigned int */
+                                      ucs_min(mm_config->fifo_max_poll, UINT_MAX));
+    self->fifo_poll_window_size    = self->config.fifo_max_poll;
+    self->fifo_window_update_iter  = 0;
     /* cppcheck-suppress internalAstError */
     self->fifo_release_factor_mask = UCS_MASK(ucs_ilog2(ucs_max((int)
                                      (mm_config->fifo_size * mm_config->release_fifo_factor),
                                      1)));
-    self->fifo_mask                = mm_config->fifo_size - 1;
+    self->fifo_mask                = self->config.fifo_size - 1;
     self->fifo_shift               = ucs_count_trailing_zero_bits(mm_config->fifo_size);
     self->rx_headroom              = (params->field_mask &
                                       UCT_IFACE_PARAM_FIELD_RX_HEADROOM) ?
                                      params->rx_headroom : 0;
     self->release_desc.cb          = uct_mm_iface_release_desc;
+    /* check that the fifo size, from the user, is a power of two and bigger than 1 */
+    if (mm_config->fifo_window_update_max_iter == UCS_ULUNITS_AUTO) {
+        self->config.fifo_window_update_max_iter = 128;
```

Comment:
better use constant (and `UCS_PP_MAKE_STRING` when defining it for env var)

Reply:
done